x-pack/legacy/plugins/apm/public/components/app/Settings/AddSettings/AddSettingFlyout.tsx

```diff
@@ -59,8 +59,10 @@ export function AddSettingsFlyout({
   const [serviceName, setServiceName] = useState<string | undefined>(
     selectedConfig ? selectedConfig.service.name : undefined
   );
-  const [sampleRate, setSampleRate] = useState<number>(
-    selectedConfig ? parseFloat(selectedConfig.settings.sample_rate) : NaN
+  const [sampleRate, setSampleRate] = useState<string | undefined>(
+    selectedConfig
+      ? selectedConfig.settings.transaction_sample_rate.toString()
+      : undefined
```

Comment:
If you change this to:
```js
  const [sampleRate, setSampleRate] = useState(
    selectedConfig
      ? selectedConfig.settings.transaction_sample_rate.toString()
      : ''
```

won't that solve some issues downstream where you have to check whether `sampleRate` is defined and then default to `''`?